packages/drivers/odsp-driver/src/odspCache.ts

```diff
@@ -233,11 +242,16 @@ export class LocalPersistentCache implements IPersistedCache {
     updateUsage(entry: ICacheEntry, seqNumber: number): void {
     }
 
-    async removeAllEntriesForDocId(docId: string): Promise<void> {
+    async removeEntries(file?: IFileEntry, type?: CacheContentType, key?: string): Promise<void> {
         Array.from(this.cache)
-        .filter(([key]) => key.startsWith(docId))
-        .map(([key]) => {
-            this.cache.delete(key);
+        .filter(([cachekey]) => {
+            const keyToBeDeleted = `${file?.docId}_${type ? `${type}_${key ? key : ""}` : ""}`;
```

Comment:
This does not look right for case where docId & key are supplied, but type is not supplied.
I'd say - let's remove key for now from API, we can always add it later

Reply:
Yes, I am still working on the key.